e2e/LongHaul/service/IotHub.cs

```diff
@@ -184,17 +200,48 @@ AcknowledgementType OnC2dMessageAck(FeedbackBatch feedbackMessages)
             }
             finally
             {
-                await s_serviceClient.MessageFeedback.CloseAsync().ConfigureAwait(false);
+                await _serviceClient.MessageFeedback.CloseAsync().ConfigureAwait(false);
+            }
+        }
+
+        public async Task ReceiveFileUploadAsync(CancellationToken ct)
+        {
+            while (!ct.IsCancellationRequested)
+            {
+                await _serviceClient.FileUploadNotifications.OpenAsync(ct).ConfigureAwait(false);
+                _logger.Trace("Listening for file upload notifications from the service...");
+
+                AcknowledgementType FileUploadNotificationCallback(FileUploadNotification fileUploadNotification)
+                {
+                    AcknowledgementType ackType = AcknowledgementType.Complete;
+                    _totalFileUploadNotificationsReceived++;
+
+                    var sb = new StringBuilder();
+                    sb.Append($"Received file upload notification.");
+                    sb.Append($"\tDeviceId: {fileUploadNotification.DeviceId ?? "N/A"}.");
+                    sb.Append($"\tFileName: {fileUploadNotification.BlobName ?? "N/A"}.");
+                    sb.Append($"\tEnqueueTimeUTC: {fileUploadNotification.EnqueuedOnUtc}.");
+                    sb.Append($"\tBlobSizeInBytes: {fileUploadNotification.BlobSizeInBytes}.");
+                    _logger.Trace(sb.ToString());
+
+                    _blobContainerClient.DeleteBlobIfExists(fileUploadNotification.BlobName);
+                    return ackType;
+                }
+
+                _serviceClient.FileUploadNotifications.FileUploadNotificationProcessor = FileUploadNotificationCallback;
+                _logger.Metric("TotalFileUploadNotificiationsReceived", _totalFileUploadNotificationsReceived);
+
+                await Task.Delay(30 * 1000);
```

Comment:
This would look better as a timespan. You can even declare the timespan above the while loop so it doesn't get recreated each loop iteration.

```suggestion
                await Task.Delay(TimeSpan.FromSeconds(30));
```

Reply:
Use a variable to specify the interval, similar to `s_c2dMessagesSentInterval`